### ISSUE TITLE
Is it possible to return a test as plain XML text?

### ISSUE THREAD

**Issue by Taomyn** [librespeed/speedtest#420]
I have librespeed all nicely set up and working well as a docker container, actually a podman container on Fedora.

What I'd like to be able to do is use wget or curl from any other machine and get back test results as XML - then I can use Zabbix to poll various systems to check the connection speeds into on central place. Is there someway to achieve this with the current implementation?

**Comment by adolfintel** [comment]
results/index.php currently only outputs the data as image, but it could be adapted to output json (that's what I'd recommend) or even XML if you prefer it.

**Comment by adolfintel** [comment]
I just added results/json.php to the repo for your use case.

**Comment by Taomyn** [comment]
> I just added results/json.php to the repo for your use case.

Wow, thank-you - I will give it a try just as soon as I can.

**Comment by Taomyn** [comment]
@adolfintel doesn't quite work as I expected, thought it would run the test and then return the results of that test. All I get is empty data so I presume it needs me to give it the test ID, but as no test has been done I would not have that.

**Comment by adolfintel** [comment]
Yes, it expects to receive the test ID. Do you want it to just dump the database?

**Comment by Taomyn** [comment]
> Yes, it expects to receive the test ID. Do you want it to just dump the database?

Nope, I was hoping it would run the test then return its results, nothing from the database i.e. do the same as a user hitting "Start" but without the GUI and just returning the data as JSON/XML

**Comment by adolfintel** [comment]
You can use the [CLI version](https://github.com/librespeed/speedtest-cli) for that, it can output as JSON.

**Comment by Taomyn** [comment]
> You can use the [CLI version](https://github.com/librespeed/speedtest-cli) for that, it can output as JSON.

That means having to have it on each machine to be tested, the correct version of the OS etc etc. Being able to run the test with a simple wget/curl command that even Windows can execute saves all that hassle.

**Comment by adolfintel** [comment]
So you want something like a service that runs on a machine, and when you connect to it, it runs the test from that machine to a server, and then forward the results to you?

**Comment by Taomyn** [comment]
No nothing like that. Basically a page on the librepseed server much like you have just done with json.php, but it first runs the test before sending back the results as just text.

For example, on any machine with access to my librespeed server that supported wget I would just type:

`wget http://librespeed:888/results/json.php`

It would run the test and return the test result, in the format you already did. Whatever was running the wget can then scrape or pickup the result if saved to a file.

**Comment by adolfintel** [comment]
So you want to wget a page and it does a test from you to the server that you're fetching the page from?

**Comment by Taomyn** [comment]
Yes please :-)

**Comment by adolfintel** [comment]
You can't do that with wget, it needs to run javascript on the client to do the test. The closest thing we have is the CLI version.

**Comment by Taomyn** [comment]
Ah, of course, what was I thinking, but there might be a way as I've done something to run javascript from the command-line some time back. Hmmmmm.

Thanks again for the help anyway.

**Comment by adolfintel** [comment]
Just use the CLI version, it does exactly what you're looking for, and you don't even have to install it, the binaries are statically built.

**Comment by Taomyn** [comment]
> Just use the CLI version, it does exactly what you're looking for, and you don't even have to install it, the binaries are statically linked.

Not sure now: https://github.com/librespeed/speedtest-cli/issues/29
